crate-tmp/src/nvtxw.rs

```diff
@@ -0,0 +1,241 @@
+pub const NVTX_PAYLOAD_SCHEMA_TYPE_DYNAMIC: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_TYPE_DYNAMIC as u64;
+
+pub const NVTX_PAYLOAD_SCHEMA_FLAG_NONE: u64 = nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_FLAG_NONE as u64;
+pub const NVTX_PAYLOAD_SCHEMA_FLAG_REFERENCED: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_FLAG_REFERENCED as u64;
+
+pub const NVTX_PAYLOAD_SCHEMA_ATTR_NAME: u64 = nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_ATTR_NAME as u64;
+pub const NVTX_PAYLOAD_SCHEMA_ATTR_TYPE: u64 = nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_ATTR_TYPE as u64;
+pub const NVTX_PAYLOAD_SCHEMA_ATTR_FLAGS: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_ATTR_FLAGS as u64;
+pub const NVTX_PAYLOAD_SCHEMA_ATTR_ENTRIES: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_ATTR_ENTRIES as u64;
+pub const NVTX_PAYLOAD_SCHEMA_ATTR_NUM_ENTRIES: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_ATTR_NUM_ENTRIES as u64;
+pub const NVTX_PAYLOAD_SCHEMA_ATTR_STATIC_SIZE: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_ATTR_STATIC_SIZE as u64;
+pub const NVTX_PAYLOAD_SCHEMA_ATTR_SCHEMA_ID: u64 =
+    nvtxw_bindings::NVTX_PAYLOAD_SCHEMA_ATTR_SCHEMA_ID as u64;
+
+const NVTXW3_STREAM_ORDER_INTERLEAVING_NONE: i16 = 0;
+const NVTXW3_STREAM_ORDERING_TYPE_UNKNOWN: i16 = 0;
+
+fn check(result: ResultCode) -> Result<(), ResultCode> {
+    if result != NVTXW3_RESULT_SUCCESS {
+        return Err(result);
+    }
+    Ok(())
+}
+
+pub fn initialize_simple() -> Result<InterfaceHandle, ResultCode> {
+    let so_name = env::var("NVTXW_BACKEND").unwrap_or_else(|_| "libNvtxwBackend.so".to_string());
+    let c_mode_string = CString::new(so_name).expect("c_mode_string CString::new failed");
+
+    let mut get_interface_func: GetInterface = Default::default();
+    check(unsafe {
+        nvtxw_bindings::nvtxwInitialize(
+            NVTXW3_INIT_MODE_LIBRARY_FILENAME,
+            c_mode_string.as_ptr(),
+            &mut get_interface_func,
+            null_mut(),
+        )
+    })?;
+
+    let get_interface = get_interface_func.ok_or(NVTXW3_RESULT_FAILED)?;
+
+    let mut interface_void: *const c_void = null();
+    check(unsafe { get_interface(NVTXW3_INTERFACE_ID_CORE_V1, &mut interface_void) })?;
+
+    if interface_void.is_null() {
+        return Err(NVTXW3_RESULT_FAILED);
+    }
+
+    let ptr_interface = interface_void as *const InterfaceHandle;
+    let interface;
+    unsafe {
+        interface = *ptr_interface;
+    }
+
+    Ok(interface)
+}
+
+pub fn session_begin_simple(
+    interface: InterfaceHandle,
+    output: OsString,
+    merge: Option<OsString>,
+) -> Result<SessionHandle, ResultCode> {
+    let session_begin = interface
+        .SessionBegin
+        .ok_or(NVTXW3_RESULT_INVALID_ARGUMENT)?;
+
+    let mut path_output = PathBuf::from(output);
+    // avoid "*.nsys-rep.nsys-rep"
+    if path_output.extension() == Some(OsStr::new("nsys-rep")) {
+        path_output.set_extension("");
+    }
+    let s_output = path_output
+        .as_os_str()
+        .to_str()
```

Comment:
I think you can do `.to_str()` on a `PathBuf` directly so you can skip the indirection through `.as_os_str()`:

```suggestion
        .to_str()
```

Reply:
Fixed